src/view_dna_n_to_random.hpp

```diff
@@ -29,6 +30,7 @@
 #include <seqan3/range/views/deep.hpp>
 #include <seqan3/std/ranges>
 
+template <typename TAlph>
```

Comment:
```suggestion
template <seqan3::nucleotide_alphabet TAlph>
```

Maybe restrict to only nucleotides? Otherwise, 'N' has no clear meaning.